ws/hub.go

```diff
@@ -51,7 +62,25 @@ func (h *Hub) run() {
 				close(client.send)
 			}
 		case rec := <-h.broadcast:
-			msg := encodeRecord(rec.Value())
+			rid := rec.Value().Cid().String()
+
+			ctx, cancel := context.WithTimeout(h.ctx, encodeRecordTimeout)
+			jrec, err := recordToJSON(ctx, h.service, rec.Value())
+			if err != nil {
+				log.Errorf("error converting record %s to JSON: %v", rid, err)
+				break
+			}
+			cancel()
```

Comment:
maybe moving to a `defer cancel()` after `.WithTimeout` to cover the case of `err != nil`.

Reply:
`defer` here is in a blocking for loop - wouldn't that lead to a resource leak?

Reply:
Ops, yes, didn't notice the outer loop. So then just adding `cancel()`  before `break`.